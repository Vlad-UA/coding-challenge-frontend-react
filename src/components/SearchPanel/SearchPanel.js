// Core
import React, { PureComponent } from 'react';
import moment from 'moment';

// Instruments
import styles from './SearchPanel.module.scss';

export default class SearchPanel extends PureComponent {
  state = {
    // TODO remove from value
    dateFrom: moment('2018-01-01').format('YYYY-MM-DD'),
    dateTo: moment().format('YYYY-MM-DD'),
    query: '',
  };

  onChangeData = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = (event) => {
    const { onFind } = this.props;
    const { dateFrom, dateTo, query } = this.state;

    let isError = false;

    event.preventDefault();

    if (!moment(dateFrom).isValid()) {
      isError = true;
      console.log('dateFrom is not valid', dateFrom);
    }

    if (!moment(dateTo).isValid()) {
      isError = true;
      console.log('dateTo is not valid', dateTo);
    }

    if (!isError) {
      onFind({
        dateFrom: moment(dateFrom).unix(),
        dateTo: moment(dateTo).unix(),
        query,
      });
    }
  };

  render() {
    const { isDataLoading } = this.props;
    const { dateFrom, dateTo } = this.state;


    return (
      <div className={styles.SearchPanel}>
        <form onSubmit={this.onSubmit}>
          <input type="text" placeholder="Search case descriptions" name="query" maxLength={30} onChange={this.onChangeData} />
          <div>
          From
            <input type="date" placeholder="from" name="dateFrom" onChange={this.onChangeData} value={dateFrom} />
          </div>
          <div>
          To
            <input type="date" placeholder="to" name="dateTo" onChange={this.onChangeData} value={dateTo} />
          </div>

          <button type="submit" disabled={isDataLoading} onClick={this.onSubmit}>Find cases</button>
        </form>
      </div>
    );
  }
}

// TODO add propTypes
