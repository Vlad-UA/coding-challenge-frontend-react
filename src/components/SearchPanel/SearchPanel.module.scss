.SearchPanel {
  form{
    text-align: center;

    .inline {
      display: inline-block;
      margin-right: 10px;
      margin-top: 20px;
    }

    .date {
      display: inline-block;
      margin-right: 10px;
    }
  }

  .errorMessage {
    color: red;
    font-weight: bold;
  }
}
