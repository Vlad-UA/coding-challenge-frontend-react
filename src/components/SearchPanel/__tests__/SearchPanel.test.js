// Core
import React from 'react';

// Test utilities
import renderer from 'react-test-renderer';

// Component
import SearchPanel from '../SearchPanel';

describe('Component SearchPanel:', () => {
  test('Must match snapshot:', () => {
    const searchPanel = renderer
      .create(<SearchPanel isDataLoading={false} onFind={jest.fn} />)
      .toJSON();

    expect(searchPanel).toMatchSnapshot();
  });
});
