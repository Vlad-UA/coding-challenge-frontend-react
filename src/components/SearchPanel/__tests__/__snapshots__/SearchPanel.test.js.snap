// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SearchPanel: Must match snapshot: 1`] = `
<div
  className="SearchPanel"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="inline"
      maxLength={30}
      name="query"
      onChange={[Function]}
      placeholder="Search case descriptions"
      type="text"
    />
    <div
      className="inline"
    >
      <div
        className="date"
      >
        <div>
          From
        </div>
        <input
          name="dateFrom"
          onChange={[Function]}
          placeholder="from"
          type="date"
        />
      </div>
      <div
        className="date"
      >
        <div>
          To
        </div>
        <input
          name="dateTo"
          onChange={[Function]}
          placeholder="to"
          type="date"
        />
      </div>
    </div>
    <button
      className="inline"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Find cases
    </button>
  </form>
</div>
`;
